metrotranscom/doorway
# Release/2024 02 23

# Pull Request Template

## Issue Overview

This PR addresses [#389](https://github.com/bloom-housing/bloom/issues/3893)


## Description

This PR pulls over the prisma changes from core and also has the majority of the changes specific to Doorway.

*Note* working through the following still so the PR isn't 100% ready, but can be looked at by the team to find any issues.

What is still remaining:
- ~Filtering on the listings page~
- ~Tests working in github actions~
- ~Updating the docker images for backend~
- Pull in the mandated account changes

## How Can This Be Tested/Reviewed?

This can only be tested locally and all flows on both public/partner sites should be tested (excluding the items in "what is still remaining" above)

Notes to test this locally:
- Make sure you have a `.env` file in api directory with the values from `.env.template`. You can use the same values you might have setup in core. But note there are a few values that are specific to Doorway such as the values to publish to s3. If you don't already have those values reach out to Morgan
- The best way to get setup is to do the following:
  - run `yarn install` from api directory
  - run `yarn prisma generate` from api directory
  - run `yarn db:reseed:with-external:dev` from api directory
  - run `yarn dev:all` from root directory (make sure you have also installed at root)

Specific areas that should be tested extra closely:
- Adding images to a listing (make sure that the aws flow works)
- Filtering on the public listings page. This was rewritten so making sure that no functionality on the listing page is broken


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
